### PR TITLE
first commit of challenges

### PR DESCRIPTION
**Pull request by arielMKS**

### CODE REVIEW

File: js/challenge-1/solutions/arielsolution2.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+function foo(numsArray) {
```
Comment (jcampos8782) [inline js/challenge-1/solutions/arielsolution2.js]: As we discussed earlier, this is a really nice solution but it doesn't work with zeroes. Try a couple different test cases in your console statements. Try with one or more zeroes.

**Comment by arielMKS** [comment]
Cool, thanks for reviewing!

Regarding edge case (zeroes), version #1 actually works but NOT version #2.
However, I think version #2 can be change a bit so that with zero
element(s) in the array the output array will be mostly zeros except the
last element. For example, input of [0,2,3,6] should return output of
[0,0,0,36]. I'll modify and push my code.

Thanks Jason!

File: js/challenge-1/solutions/arielsolution3.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  // handle multiple zero elements in array
+  const [a, b] = numsArray;
+  if (a + b === 0) {
+    return Array(numsArray.length).fill(0);
+  }
+
+  // handle zero element in array
+  if (a === 0) {
+    result = Array(numsArray.length - 1).fill(0); // create new array fill with 0
+    result.push(numsArray[numsArray.length - 1] ** 2); // push last element**2 at end of array;
```
Comment (jcampos8782) [inline js/challenge-1/solutions/arielsolution3.js]: You can't just square the last element I don't think. This works in this case because the other items in the array are 2 and 3... 2 * 3 just happens to equal 6.

File: js/challenge-1/solutions/arielsolution3.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+function foo(numsArray) {
+  // assume numsArray is always sorted
```
Comment (jcampos8782) [inline js/challenge-1/solutions/arielsolution3.js]: Not a fair assumption based on the stated input in the problem. You could add an array.sort though.

File: js/challenge-1/solutions/arielsolution3.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  let result;
+
+  // handle multiple zero elements in array
+  const [a, b] = numsArray;
+  if (a + b === 0) {
```
Comment (jcampos8782) [inline js/challenge-1/solutions/arielsolution3.js]: What if the first two elements are [-1,1] or [-2,2] etc ?

File: js/challenge-2/solutions/arielsolution1.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    z: 0
+  };
+
+  for (let i = 0; i < lowercase.length; i++) {
+    if (obj.hasOwnProperty(lowercase[i])) {
+      obj[lowercase[i]]++;
+    }
+  }
+
+  return Object.values(obj);
```
Comment (jcampos8782) [inline js/challenge-2/solutions/arielsolution1.js]: Interesting implementation. It definitely meets the requirements but I think you can do this without needing an intermediate letter map. See https://www.techonthenet.com/js/string_charcodeat.php

**Comment by arielMKS** [comment]
Cool thank you Jason!

**Comment by jcampos8782** [comment]
Closing for now til the issues in the comments are resolved.
